src/test/java/com/fasterxml/jackson/databind/tofix/DeserializationWithCreatorPropertyOrder3355Test.java

```diff
@@ -0,0 +1,80 @@
+package com.fasterxml.jackson.databind.tofix;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.testutil.DatabindTestUtil;
+import com.fasterxml.jackson.databind.testutil.failure.JacksonTestFailureExpected;
+import org.junit.jupiter.api.Test;
```

Comment:
Going forward, let's remember to order this as per coding style (above before Jackson imports).
I'll fix